src/Database/LSMTree/Internal.hs

```diff
@@ -868,12 +877,17 @@ newCursor th = withOpenTable th $ \thEnv -> do
     -- references to each run, so it is safe.
     allocTableContent reg contentVar = do
         RW.withReadAccess contentVar $ \content -> do
+          let wb      = tableWriteBuffer content
+              wbblobs = tableWriteBufferBlobs content
+          allocateTemp reg
+            (WBB.addReference wbblobs)
+            (\_ -> WBB.removeReference wbblobs)
```

Comment:
Ok, so if I've understood our temp registry properly, then the code above will add the reference, and then it will only remove the reference if an exception is thrown subsequently but within the scope of `withTempRegistry`. So in other words this will deal with the allocation and failure case, but not the happy case.

So presumably for that we should include the wbblobe into the cursor itself, and release it in `closeCursor`.

Reply:
Exactly!

Reply:
I applied a fix, and it also fixes the failures on Windows, which I think were caused by exactly this problem.

It may also solve the problem I was seeing with running out of file descriptors when running more tests and larger test cases.